### DISCUSSION TITLE
IModel Props Export

### DISCUSSION THREAD

**Discussion by zhang-zimin** [iTwin/itwinjs-core#7440] [Q&A] [answered]
Does ITwin.js have a method to export all property information of an IModel? Or is there an interface available to query it?

**Comment by pmconne** [comment, marked as answer]
Hi @zhang-zimin.

iTwin.js provides an entire [set of libraries](https://www.itwinjs.org/presentation/) to query and format property information. For lower-level raw property data you can use methods on [IModelConnection.elements](https://www.itwinjs.org/reference/core-frontend/imodelconnection/imodelconnection/imodelconnection.elements/). We do not provide a generic "dump all property data to <xml|csv|some-other-format>" API, but you could write one specific to your needs using the APIs I mentioned above.

**Reply by zhang-zimin** [reply]
Many thanks.
